components/match2/wikis/splitgate/match_group_input_custom.lua

```diff
@@ -0,0 +1,553 @@
+	match.extradata = {
+		matchsection = Variables.varDefault('matchsection'),
+		lastgame = Variables.varDefault('last_game'),
+		comment = match.comment,
+		mapveto = matchFunctions.getMapVeto(match),
+		mvp = matchFunctions.getMVP(match),
+		isconverted = 0
+	}
+	return match
+end
+
+-- Parse the mapVeto input
+function matchFunctions.getMapVeto(match)
+	if not match.vetoes then return {} end
+
+	local vetoes = mw.text.split(match.vetoes or '', ',')
+	match.vetoes = nil
+	local vetoesBy = mw.text.split(match.vetoesBy or '', ',')
+	match.vetoesBy = nil
+	local index = 1
+	local currentVetoMap = mw.text.trim(vetoes[1])
+	local vetoData = {}
+
+	while not String.isEmpty(currentVetoMap) do
+		local by = tonumber(mw.text.trim(vetoesBy[index]) or '')
+		vetoData[index] = { map = currentVetoMap, by = by }
+		index = index + 1
+		currentVetoMap = mw.text.trim(vetoes[index] or '')
+	end
+
+	return vetoData
+end
+
+-- Parse MVP input
+function matchFunctions.getMVP(match)
+	if not match.mvp then return {} end
+	local mvppoints = match.mvppoints or 1
+
+	-- Split the input
+	local players = mw.text.split(match.mvp, ',')
+
+	-- Trim the input
+	for index,player in pairs(players) do
+		players[index] = mw.text.trim(player)
+	end
+
+	return {players=players, points=mvppoints}
+end
+
+function matchFunctions.getOpponents(match)
+	-- read opponents and ignore empty ones
+	local opponents = {}
+	local isScoreSet = false
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		-- read opponent
+		local opponent = match['opponent' .. opponentIndex]
+		if not Logic.isEmpty(opponent) then
+			--retrieve name and icon for teams from team templates
+			if opponent.type == 'team' and
+				not Logic.isEmpty(opponent.template, match.date) then
+					local name, icon, template = opponentFunctions.getTeamNameAndIcon(opponent.template, match.date)
+					opponent.template = template or opponent.template
+					opponent.name = opponent.name or name
+					opponent.icon = opponent.icon or icon
+			end
+			if not String.isEmpty(opponent.name) then
+				opponent.name = mw.ext.TeamLiquidIntegration.resolve_redirect(opponent.name)
+			end
+
+			-- apply status
+			if TypeUtil.isNumeric(opponent.score) then
+				opponent.status = 'S'
+				isScoreSet = true
+			elseif Table.includes(ALLOWED_STATUSES, opponent.score) then
+				opponent.status = opponent.score
+				opponent.score = -1
+			end
+			opponents[opponentIndex] = opponent
+
+			-- get players from vars for teams
+			if opponent.type == 'team' and not Logic.isEmpty(opponent.name) then
+				match = matchFunctions.getPlayers(match, opponentIndex, opponent.name)
+			end
+		end
+	end
+
+	-- see if match should actually be finished if bestof limit was reached
+	if isScoreSet and not Logic.readBool(match.finished) then
+		local firstTo = math.ceil(match.bestof/2)
+		for _, item in pairs(opponents) do
+			if tonumber(item.score or 0) >= firstTo then
+				match.finished = true
+				break
+			end
+		end
+	end
```

Comment:
```suggestion
	end
	-- check if match should actually be finished due to a non score status
	if not Logic.readBool(match.finished) then
		for _, item in pairs(opponents) do
			if String.isNotEmpty(opponent.status) and opponent.status ~= 'S'
				match.finished = true
				break
			end
		end
	end
```

Reply:
Code as is throws an error on line 441 for expecting a then after the if statement but finding none. Code added with a 'then' after 'S' on line 439

Reply:
yeah missed a `then` in the suggestion^^